File: app/src/main/java/com/hhh/onepiece/model/WorksModel.java
package com.hhh.onepiece.model;

import androidx.annotation.NonNull;

import com.hhh.mvvm.listener.DiffCallback;

public final class WorksModel implements DiffCallback<WorksModel> {

  public int mId;

  public int mColor;

  public float mAspectRatio;

  public String mText;

  public transient boolean mShown;

  @Override
  public boolean areItemsTheSame(@NonNull WorksModel newItem) {
    return mId == newItem.mId;
  }

  @Override
  public boolean areContentsTheSame(@NonNull WorksModel newItem) {
    return areItemsTheSame(newItem) && mColor == newItem.mColor;
  }
}

File: app/src/main/java/com/nb/core/test/model/WorksModel.kt
package com.nb.core.test.model

import com.nb.core.DiffCallback

data class WorksModel(val id: Int, val color: Int, val aspectRatio: Float, val text: String) :
    DiffCallback<WorksModel> {

    override fun areItemsTheSame(newItem: WorksModel): Boolean {
        return id == newItem.id
    }

    override fun areContentsTheSame(newItem: WorksModel): Boolean {
        return areItemsTheSame(newItem) && color == newItem.color
    }

}
